index.d.ts

```diff
@@ -0,0 +1,131 @@
+
+declare namespace TwitterLite {
+  interface TwitterOptions {
+    /** "api" is the default (change for other subdomains) */
+    subdomain?: string;
+    /** version "1.1" is the default (change for other subdomains) */
+    version?: string;
+    /** consumer key from Twitter. */
+    consumer_key: string;
+    /** consumer secret from Twitter */
+    consumer_secret: string;
+    /** access token key from your User (oauth_token) */
+    access_token_key?: OauthToken;
+    /** access token secret from your User (oauth_token_secret) */
+    access_token_secret?: OauthTokenSecret;
+  }
+
+  type OauthToken = string;
+  type OauthTokenSecret = string;
+  type AuthType = 'App' | 'User';
+
+  interface KeySecret {
+    key: string;
+    secret: string;
+  }
+
+  interface AccessTokenOptions extends KeySecret {
+    /** If using the OAuth web-flow, set this parameter to the value of the oauth_verifier returned in the callback URL. If you are using out-of-band OAuth, set this value to the pin-code. */
+    verifier: string | number;
+  }
+
+  interface BearerResponse {
+    token_type: 'bearer';
+    access_token: string;
+  }
+
+  interface TokenResponse {
+    oauth_token: OauthToken;
+    oauth_token_secret: OauthTokenSecret;
+  }
+
+  interface AccessTokenResponse extends TokenResponse {
+    user_id: number;
+    screen_name: string;
+  }
+
+  class Twitter {
+    #authType: AuthType;
+    #url: string;
+    #oauth: string;
+    #config: TwitterOptions;
+    #client: OAuth;
+    #token: KeySecret;
+
+    constructor(options: TwitterOptions);
+  
+    /**
+     * Parse the JSON from a Response object and add the Headers under `_headers`
+     */
+    private static _handleResponse(response: Response): Promise<object>;
+  
+    getBearerToken(): Promise<BearerResponse>;
+  
+    /** The value you specify here will be used as the URL a user is redirected to should they approve your application's access to their account. Set this to oob for out-of-band pin mode. */
+    getRequestToken(twitterCallbackUrl: string | 'oob'): Promise<TokenResponse>;
+
+    getAccessToken(options: AccessTokenOptions): Promise<AccessTokenResponse>;
+  
+    /**
+     * Construct the data and headers for an authenticated HTTP request to the Twitter API
+     * @param {string} resource - the API endpoint
+     */
+    private _makeRequest(method: 'GET' | 'POST' | 'PUT', resource: string, parameters: object): { requestData: { url: string; method: string; }; headers: ({ Authorization: string; } | OAuth.Header); };
+
+  
+    /**
+     * Send a GET request
+     * @param {string} resource - endpoint, e.g. `followers/ids`
+     * @param {object} [parameters] - optional parameters
+     * @returns {Promise<object>} Promise resolving to the response from the Twitter API.
+     *   The `_header` property will be set to the Response headers (useful for checking rate limits)
+     */
+    public get(resource: string, parameters?: object): Promise<Object>;
```

Comment:
Should the return type here be `Promisie<Object> | Promise<Array>`? Someone pointed out that some endpoints might return an array, so we should keep that in mind. If this is not a problem though we can merge, but it seems like it could be.

Reply:
Tbh making it Promise<any> might be the best bet, because currently it *could be* anything. This also implies that its not really typed there. Only problem would be of projects don't allow any, but maybe they can fix this by immediately typecasting. IDK how any from libraries would be handled by that TS rule

Reply:
*`Promise<any>` instead of Promise

Reply:
I agree with @T0TProduction, going with `Promise<any>` for now seems like the best option, especially if we're going to add overloads for each endpoint. Those overloads should be the subject for a new PR though, so that people relying on typings can start working with these 'basic' ones.

Reply:
I disagree with ```Promise<any>```, it would be better to just do
```public get<T>(resource: string, parameters?: object): Promise<T>;```
for the time being and let the user pass the expected result instead of casting it in his code.

Reply:
good point. Will T default to any if not specified? not sure about that behavior. If so, I agree.

Reply:
I reckon  ` public get<T = any>(resource: string, parameters?: object): Promise<T>;` would provide that behaviour.

Reply:
Passing a generic is overall more user friendly and correct then explicit casting

Reply:
Ammended the PR with your suggestions. What do you think @naorzr?

Reply:
wouldn't 
```public get<T>(resource: string, parameters?: object): Promise<T>;```
cause the transpiler to complain in case it doesn't get a generic?
I think it's better to just test it out, if the transpiler doesn't complain then leave it as it is, if it does, you can just default it with
```public get<T = any>(resource: string, parameters?: object): Promise<T>;```
Other than that, looks good, thanks for the efforts!

Reply:
It becomes `unknown`, which seems undesirable. I'll add `any` as a default to the type params.